packages/network/src/types.js

```diff
@@ -11,53 +18,53 @@
 
 /**
  * @typedef {object} Closable A closable object
- * @property {() => Promise<void>} close Terminate the object
+ * @property {() => PromiseWhenable<void>} close Terminate the object
  */
 
 /**
  * @typedef {object} Protocol The network Protocol
- * @property {(prefix: Endpoint) => Promise<Port>} bind Claim a port, or if
+ * @property {(prefix: Endpoint) => PromiseWhenable<Port>} bind Claim a port, or if
  *   ending in ENDPOINT_SEPARATOR, a fresh name
  */
 
 /**
  * @typedef {object} Port A port that has been bound to a protocol
  * @property {() => Endpoint} getLocalAddress Get the locally bound name of this
  *   port
- * @property {(acceptHandler: ListenHandler) => Promise<void>} addListener
+ * @property {(acceptHandler: ListenHandler) => PromiseWhenable<void>} addListener
```

Comment:
In https://github.com/Agoric/agoric-sdk/pull/8742#issuecomment-1924441086 , @michaelfig writes:
> The case of the Network API is a special one, because it deals with promises that are usually long-lived. I don't expect short-lived promises to need Whenable support in general.

Is the promise returned from `addListener` expected to be long-lived?
Likewise `close`, `bind`, etc. It seems like most of the methods in this API should not need Whenable support.

Reply:
Discussed offline. It's a great point that a lot of these don't need to be long-lived. The current approach is to have all of these return a `Vow` and let the consumer decide what to do with it. In a follow up PR, we may scope this down to just the critical APIs that need long-lived support

Reply:
let's think more about that in #8867